lib/rest-api.php

```diff
@@ -338,6 +338,48 @@ function gutenberg_add_target_schema_to_links( $response, $post, $request ) {
 	return $response;
 }
 
+/**
+ * Include revisions data on post response links.
+ *
+ * @see https://core.trac.wordpress.org/ticket/44321
+ *
+ * @param WP_REST_Response $response WP REST API response of a post.
+ * @param WP_Post          $post The post being returned.
+ * @param WP_REST_Request  $request WP REST API request.
+ * @return WP_REST_Response Response containing the new links.
+ */
+function gutenberg_add_revisions_data_to_links( $response, $post, $request ) {
+
+	$new_links  = array();
+	$orig_links = $response->get_links();
+
+	if ( in_array( $post->post_type, array( 'post', 'page' ), true ) || post_type_supports( $post->post_type, 'revisions' ) ) {
```

Comment:
Why do we hardcode support for `post` and `page` types, rather than letting it follow the same logic of using `post_type_supports` (since, by default, they do register support)? If it's needed, can we leave an inline comment explaining why?

Reply:
> Why do we hardcode support for `post` and `page `types, rather than letting it follow the same logic of using `post_type_supports` (since, by default, they do register support)?

To "force" Posts and Pages to include the revisions endpoint (even if a developer has removed support via `post_type_supports`). This was a decision made in the original API design. The conditional is copied directly.

However, this is moot with 874d8b79e